apps/typegpu-docs/src/content/examples/algorithms/mnist-inference/index.ts

```diff
@@ -0,0 +1,390 @@
+import { type F32, type TgpuArray, arrayOf, f32 } from 'typegpu/data';
+import tgpu, { type TgpuBuffer, type Storage } from 'typegpu/experimental';
+
+const canvas = document.querySelector('canvas') as HTMLCanvasElement;
+
+const root = await tgpu.init();
+const device = root.device;
+const data = new Uint8Array(28 * 28);
+
+// Shader code
+
+const layerShader = `
+  @binding(0) @group(0) var<storage, read> input: array<f32>;
+  @binding(1) @group(0) var<storage, read_write> output: array<f32>;
+
+  @binding(0) @group(1) var<storage, read> weights: array<f32>;
+  @binding(1) @group(1) var<storage, read> biases: array<f32>;
+
+  fn relu(x: f32) -> f32 {
+    return max(0.0, x);
+  }
+
+  @compute @workgroup_size(1)
+  fn main(@builtin(global_invocation_id) gid: vec3u) {
+    let inputSize = arrayLength( &input );
+
+    let i = gid.x;
+
+    let weightsOffset = i * inputSize;
+    var sum = 0.0;
+
+    for (var j = 0u; j < inputSize; j = j + 1) {
+      sum = sum + input[j] * weights[weightsOffset + j];
+    }
+
+    sum = sum + biases[i];
+    output[i] = relu(sum);
+  }
+`;
+
+const inputOutput = tgpu.bindGroupLayout({
+  input: {
+    storage: (length: number) => {
+      return arrayOf(f32, length);
+    },
+    access: 'readonly',
+  },
+  output: {
+    storage: (length: number) => {
+      return arrayOf(f32, length);
+    },
+    access: 'mutable',
+  },
+});
+
+const weightsBiases = tgpu.bindGroupLayout({
+  weights: {
+    storage: (length: number) => {
+      return arrayOf(f32, length);
+    },
+    access: 'readonly',
+  },
+  biases: {
+    storage: (length: number) => {
+      return arrayOf(f32, length);
+    },
+    access: 'readonly',
+  },
+});
+
+const pipelineLayout = device.createPipelineLayout({
+  bindGroupLayouts: [root.unwrap(inputOutput), root.unwrap(weightsBiases)],
+});
+
+const pipeline = device.createComputePipeline({
+  layout: pipelineLayout,
+  compute: {
+    module: device.createShaderModule({
+      code: layerShader,
+    }),
+  },
+});
+
+// Definitions for the network
+
+interface LayerData {
+  header: string;
+  data: Float32Array;
+  shape: [number, number?];
+  buffer: TgpuBuffer<TgpuArray<F32>> & Storage;
+}
+
+interface Layer {
+  weights: LayerData;
+  biases: LayerData;
+  buffers: {
+    weights: TgpuBuffer<TgpuArray<F32>> & Storage;
+    biases: TgpuBuffer<TgpuArray<F32>> & Storage;
+    state: TgpuBuffer<TgpuArray<F32>> & Storage;
+  };
+}
+
+interface Network {
+  layers: Layer[];
+  input: TgpuBuffer<TgpuArray<F32>> & Storage;
+  output: TgpuBuffer<TgpuArray<F32>> & Storage;
+
+  inference(data: number[]): Promise<number[]>;
+}
+
+// Network loading functions
+
+/*
```

Comment:
Tiny nit, doc comment
```suggestion
/**
```